src/tasks/block.rs

```diff
@@ -207,38 +228,108 @@ impl BlockBuilder {
 
     /// Spawn the block builder task, returning the inbound channel to it, and
     /// a handle to the running task.
-    pub fn spawn(mut self, outbound: mpsc::UnboundedSender<InProgressBlock>) -> JoinHandle<()> {
+    pub fn spawn(
+        mut self,
+        outbound: mpsc::UnboundedSender<InProgressBlock>,
+        ru_provider: WalletlessProvider,
+    ) -> JoinHandle<()> {
         tokio::spawn(
             async move {
                 loop {
                     // sleep the buffer time
                     tokio::time::sleep(Duration::from_secs(self.secs_to_next_target())).await;
-                    tracing::info!("beginning block build cycle");
+                    info!("beginning block build cycle");
 
                     // Build a block
                     let mut in_progress = InProgressBlock::default();
                     self.get_transactions(&mut in_progress).await;
-
-                    // TODO: Implement bundle ingestion #later
-                    // self.get_bundles(&mut in_progress).await;
+                    self.get_bundles(&ru_provider, &mut in_progress).await;
 
                     // Filter confirmed transactions from the block
                     self.filter_transactions(&mut in_progress).await;
```

Comment:
realizing right now that doing this line after ingesting bundles has the _potential_ to split up bundles. 

e.g. if someone submits txn A as an individual txn; shortly after, submits a bundle with txns A, B, C; the individual txn A gets picked up and included in a block; the bundle gets picked up in a subsequent block; this logic would pick up the bundle and filter out A while including B & C, thus splitting up the bundle.

in practice this is obviously an edge case. 

but keeping the code structure like this feels like a potential footgun 

 the `filter_transactions` function as it is currently implemented essentially intended to implement a simple simulation strategy for single transactions.

Reply:
also, if we were to just remove the `filter_transactions` call, then we'd perpetually include both individual txns and bundles that have already mined until they time out in the cache

Reply:
Right, the transaction reaping is necessary, but the above would invalidate the all-or-nothing aspect of the bundle, so then instead should the submit task should:

1. pull bundles & ingest them
2. pull transactions, ingest them
2.1. filtering txs against anything already in the bundles
3. Sign and submit the block like normal 

This would prevent bundle splitting and is still the correct pattern after bundle simulation is added.